demos/guide/quic-hq-interop-server.c

```diff
@@ -0,0 +1,689 @@
+ *                  followed by intermediate CA certificates.
+ * @param key_path  Path to the server's private key file in PEM format. The
+ *                  private key must correspond to the leaf certificate in
+ *                  the chain file.
+ *
+ * @return Pointer to the initialized SSL_CTX on success, or NULL on failure.
+ *
+ * Configuration:
+ * - Loads the certificate chain and private key into the context.
+ * - Disables client certificate verification (no mutual TLS).
+ * - Sets up the ALPN selection callback for protocol negotiation.
+ *
+ * Error Handling:
+ * - If any step fails (e.g., loading the certificate or key), the function
+ *   frees the SSL_CTX and returns NULL.
+ *
+ * Usage:
+ * - Call this function to create an SSL_CTX before starting the QUIC server.
+ * - Ensure valid paths for the certificate and private key are provided.
+ *
+ * Note:
+ * - The ALPN callback only supports the predefined protocol defined in
+ *   `alpn_ossltest`.
+ */
+static SSL_CTX *create_ctx(const char *cert_path, const char *key_path)
+{
+    SSL_CTX *ctx;
+
+    /*
+     * An SSL_CTX holds shared configuration information for multiple
+     * subsequent per-client connections. We specifically load a QUIC
+     * server method here.
+     */
+    ctx = SSL_CTX_new(OSSL_QUIC_server_method());
+    if (ctx == NULL)
+        goto err;
+
+    /*
+     * Load the server's certificate *chain* file (PEM format), which includes
+     * not only the leaf (end-entity) server certificate, but also any
+     * intermediate issuer-CA certificates.  The leaf certificate must be the
+     * first certificate in the file.
+     *
+     * In advanced use-cases this can be called multiple times, once per public
+     * key algorithm for which the server has a corresponding certificate.
+     * However, the corresponding private key (see below) must be loaded first,
+     * *before* moving on to the next chain file.
+     *
+     * The requisite files "chain.pem" and "pkey.pem" can be generated by running
+     * "make chain" in this directory.  If the server will be executed from some
+     * other directory, move or copy the files there.
+     */
+    if (SSL_CTX_use_certificate_chain_file(ctx, cert_path) <= 0) {
+        fprintf(stderr, "couldn't load certificate file: %s\n", cert_path);
+        goto err;
+    }
+
+    /*
+     * Load the corresponding private key, this also checks that the private
+     * key matches the just loaded end-entity certificate.  It does not check
+     * whether the certificate chain is valid, the certificates could be
+     * expired, or may otherwise fail to form a chain that a client can validate.
+     */
+    if (SSL_CTX_use_PrivateKey_file(ctx, key_path, SSL_FILETYPE_PEM) <= 0) {
+        fprintf(stderr, "couldn't load key file: %s\n", key_path);
+        goto err;
+    }
+
+    /*
+     * Clients rarely employ certificate-based authentication, and so we don't
```

Comment:
I would drop this paragraph - it adds nothing to the demo - you are just ignoring client authentication by choice.